config/unmarshal.go

```diff
@@ -169,6 +171,58 @@ func unmarshalConfig(r io.Reader, allowEnvironmentOverrides bool) (*model.Config
 		return nil, nil, envErr
 	}
 
+	if allowEnvironmentOverrides {
+		// Making work environment variables for plugin settings
+		reg, err := regexp.Compile("[^a-zA-Z0-9]+")
+		if err != nil {
+			return nil, nil, errors.Wrapf(err, "failed to generate regular expresion")
+		}
+
+		pluginEnvs := map[string]interface{}{}
+		pluginStates := map[string]interface{}{}
+		hasPluginOverrides := false
+		hasPluginStatesOverrides := false
+		for pluginID := range config.PluginSettings.Plugins {
+			simplifiedPluginID := reg.ReplaceAllString(pluginID, "")
```

Comment:
Are we looking to remove anything else other than periods? If not, then I think we can do a simple strings.ReplaceAll of the periods here.